### PR TITLE
Promote release-level packages to a release view in internal feeds

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Fixes some issues with the logic added in #16610 that prevented it from running. This should set up proper promotion to release view for well-supported (i.e. not per-build) packages, which unblocks setting up a retention policy on Fluid's primary internal publishing feed.

### CODE REVIEW

**Comment by Abe27342** [comment]
Sample pipeline run can be found [here](https://dev.azure.com/fluidframework/internal/_build/results?buildId=221900&view=results), which ran effectively this code but with some commented out publishing stages for safety (see diff [here](https://github.com/microsoft/FluidFramework/compare/test/release-feed-publishing) and compare it to the PR'd changes).

I set up these test runs to publish to a separate feed which I'll delete after merging this PR, you can check out the status of that [here](https://dev.azure.com/fluidframework/internal/_artifacts/feed/test-absander) and confirm packages are promoted to the release view as expected.

**Comment by Abe27342** [comment]
@RishhiB -- Did work ever get filed in response to Tyler's post [here](https://github.com/microsoft/FluidFramework/pull/16610#pullrequestreview-1599056292)? If not, I'll make a repair item for the current setup, as I agree it's rather brittle.

File: tools/pipelines/templates/include-publish-npm-package-steps.yml [hunk 1]
```diff
@@ -108,23 +108,23 @@ steps:
                   exit 1
                 fi
             else
-              feedId=$(grep -oP '_packaging/\K[^/]+(?=/npm/)' publish_log)
+              feedName=$(grep -oP '_packaging/\K[^/]+(?=/npm/)' publish_log)
 
-              if [[ $release == "release" && $feedId == "build" ]]; then
+              if [[ "$(release)" == "release" && $feedName == "build" ]]; then
```
Comment (Abe27342) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:113]: This is the primary problem: `release` is an ADO variable, not a variable within the bash script. As such, this comparison never succeeded and we never ran any of this logic.

File: tools/pipelines/templates/include-publish-npm-package-steps.yml [hunk 2]
```diff
@@ -108,23 +108,23 @@ steps:
...
                 packageVersion=$(grep -oP 'npm notice version:\s+\K[^\s]+' publish_log)
 
                 if [ "${{ parameters.artifactPath }}" == "scoped" ]; then
-                  scopes=("fluidframework" "fluid-tools" "fluid-internal" "fluid-experimental")
+                  scopes=("fluidframework" "fluid-tools" "fluid-internal" "fluid-experimental" "fluid-private" "fluid-example")
```
Comment (Abe27342) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:117]: Secondary problem: we've changed up our namespaces some. fluid-example is necessary for table-document (unfortunately), and fluid-private got added.

File: tools/pipelines/templates/include-publish-npm-package-steps.yml [hunk 3]
```diff
@@ -108,23 +108,23 @@ steps:
...
                   for scope in "${scopes[@]}"; do
                       if [[ "$packageName" == "$scope"* ]]; then
                           packageName="@${packageName/$scope-/$scope/}"
                       fi
                   done
                 fi
 
-                baseURL="${${{ parameters.feedUrl }}%/registry/}"
-                feedPromotionUrl="${baseURL/_packaging/_apis/packaging/feeds}"
+                feedPromotionUrl="$(ado-feeds-api-base)/packaging/feeds/${feedName}/npm/${packageName}/versions/${packageVersion}?api-version=7.1-preview.1"
```
Comment (Abe27342) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:125]: The logic here might actually have worked out before, but IMO it was a bit hard to follow and it was easier to construct the REST API we needed to call directly rather than manipulate the registry URL.

File: tools/pipelines/templates/include-publish-npm-package-steps.yml [hunk 4]
```diff
@@ -108,23 +108,23 @@ steps:
                   exit 1
                 fi
             else
-              feedId=$(grep -oP '_packaging/\K[^/]+(?=/npm/)' publish_log)
+              feedName=$(grep -oP '_packaging/\K[^/]+(?=/npm/)' publish_log)
```
Comment (alexvy86) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:112]: I liked the intermediate state of printing `feedName` for troubleshooting purposes, since we have no other way to know what it "resolved" to; we can't look at the publish_log after the pipeline runs. `$(release)` we do print in the Parameter step, IIRC. I'd also add an echo for `parameters.artifactPath`, because we've already done a few strides towards completely removing the distinction between them for the purposes of publishing packages in pipelines; at some point I'd like to clean it up from here, so knowing what was passed could be handy.

File: tools/pipelines/templates/include-publish-npm-package-steps.yml (hunk 1 above)
Comment (alexvy86) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:113]: Probably not for this PR, but this sounds like we don't promote things published to the `test` feed, or directly to the `dev` feed? So could we eventually run into the same problem with those? Admittedly, they probably see much less volume than `build`.

**Review by alexvy86 — APPROVED** [review]
A few small things but lgtm

**Comment by RishhiB** [comment]
> @RishhiB -- Did work ever get filed in response to Tyler's post [here](https://github.com/microsoft/FluidFramework/pull/16610#pullrequestreview-1599056292)? If not, I'll make a repair item for the current setup, as I agree it's rather brittle.

It was not created, I agree it is brittle. I can make one right now
